### PR TITLE
[SPARK-12257][SQL] Non partitioned insert into a partitioned Hive table doesn't fail

### PR DESCRIPTION
**Pull request by dilipbiswal**
Added a check and raise error on missing partition values.

Currently the check is very similar to hive.

### CODE REVIEW

**Comment by cloud-fan** [comment]
ok to test

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala [hunk 1]
```diff
@@ -155,6 +155,11 @@ case class InsertIntoHiveTable(
     val partitionColumns = fileSinkConf.getTableInfo.getProperties.getProperty("partition_columns")
     val partitionColumnNames = Option(partitionColumns).map(_.split("/")).orNull
 
+    // Validate that partition values are specified for partition columns.
+    if (partitionColumnNames != null && partitionColumnNames.size > 0 && partitionSpec.size == 0) {
+      throw new SparkException(ErrorMsg.NEED_PARTITION_ERROR.getMsg)
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala]: `AnalysisException` for anything that is thrown due to an invalid query.

Comment (dilipbiswal) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala]: @marmbrus Thanks. Actually right after the code block i changed, there are a few places where we raise SparkException. So i thought there may be a reason for it and followed it. :-). I will change all those places as well.

**Comment by dilipbiswal** [comment]
@marmbrus Michael, have implemented your comments. Can you please take a look when you get a chance ?

**Comment by marmbrus** [comment]
@yhuai could you look at this?

**Comment by rxin** [comment]
Thanks for the pull request. I'm going through a list of pull requests to cut them down since the sheer number is breaking some of the tooling we have. Since most of the code in this part has been rewritten, I'm going to push a commit to close it. Feel free to reopen it or create a new one.

**Pull request closed without merging**